README.md

```diff
@@ -1,753 +1,283 @@
 
 ```js
-// File my-plugin.js
-module.exports = function myPlugin(conf) {
-  return {
-    onError: ({ inputs, netlifyConfig, constants, utils }) => {
-      console.log('Run custom logic during onError event')
-    },
-  }
+module.exports = {
+  name: 'netlify-plugin-hello-world',
+  onPreBuild: ({ constants }) => {
+    console.log(constants)
+  },
 }
 ```
 
-After creating the plugin, add into your [Netlify config](#netlify-configuration) file under the `plugins` section.
-
-Plugins can be referenced locally or installed via npm.
-
-`netlify.yml` example:
-
-```yml
-plugins:
-  - package: ./path/to/my-plugin.js
-```
-
-</details>
-
-<details>
-  <summary>Using <strong>onError</strong> via Netlify config</summary>
-  
-  <br/>
-
-Below is an example of how to use the `onError` event in the Netlify config file.
+The `constants` key contains the following values:
 
-```yml
-build:
-  lifecycle:
-    onError: echo "Do thing on onError event"
-```
-
-</details>
-
-### `onEnd`
-
-Runs on build error or success
+<!-- AUTO-GENERATED-CONTENT:START (CONSTANTS) -->
+- `CONFIG_PATH` Path to the Netlify configuration file
+- `PUBLISH_DIR` Directory that contains the deploy-ready HTML files and assets generated by the build
+- `FUNCTIONS_SRC` The directory where function source code lives
+- `FUNCTIONS_DIST` The directory where built serverless functions are placed before deployment
+- `CACHE_DIR` Path to the Netlify build cache folder
+- `IS_LOCAL` Boolean indicating whether the build was run locally (Netlify CLI) or in the production CI
+- `SITE_ID` The Netlify Site ID
+<!-- AUTO-GENERATED-CONTENT:END -->
 
-<details>
-  <summary>Using <strong>onEnd</strong> in a plugin</summary>
-  
-  <br/>
+### Error reporting
 
-Below is an example plugin using the `onEnd` event handler
+Exceptions thrown inside event handlers are reported in logs as bugs. You should handle errors with `try`/`catch` blocks
+and use `utils.build`:
 
 ```js
-// File my-plugin.js
-module.exports = function myPlugin(conf) {
-  return {
-    onEnd: ({ inputs, netlifyConfig, constants, utils }) => {
-      console.log('Run custom logic during onEnd event')
-    },
-  }
+module.exports = {
+  name: 'netlify-plugin-hello-world',
+  onPreBuild: ({ utils }) => {
+    try {
+      badMethod()
+    } catch (error) {
+      utils.build.failBuild('Failure message')
+    }
+  },
 }
 ```
 
-After creating the plugin, add into your [Netlify config](#netlify-configuration) file under the `plugins` section.
-
-Plugins can be referenced locally or installed via npm.
-
-`netlify.yml` example:
-
-```yml
-plugins:
-  - package: ./path/to/my-plugin.js
-```
-
-</details>
-
-<details>
-  <summary>Using <strong>onEnd</strong> via Netlify config</summary>
-  
-  <br/>
-
-Below is an example of how to use the `onEnd` event in the Netlify config file.
-
-```yml
-build:
-  lifecycle:
-    onEnd: echo "Do thing on onEnd event"
-```
-
-</details>
-<!-- AUTO-GENERATED-CONTENT:END (PLUGINS) -->
```

Comment:
I think the `PLUGINS` key in `scripts/docs.js` can be removed thanks to this removal.